core/src/main/java/org/elasticsearch/percolator/PercolateContext.java

```diff
@@ -154,6 +153,7 @@ public PercolateContext(PercolateShardRequest request, SearchShardTarget searchS
         this.startTime = 0;
         this.numberOfShards = 0;
         this.onlyCount = true;
+        queryShardContext = new QueryShardContext(mapperService.getIndexSettings(), null, null, null, mapperService, null, null, null);
```

Comment:
I'm wondering if the QueryShardContext should be passed into this constructor rather than created here, so the test can control how it's constructed?